npm2deb/__init__.py

```diff
@@ -415,14 +439,24 @@ def read_package_info(self):
         self._get_json_version()
         self._get_json_license()
 
-    def download(self):
+    def download_tarball(self):
         utils.debug(1, "downloading %s@%s tarball from npm registry" % (self.name, self.version))
         info = _getstatusoutput('npm pack "%s@%s"' % (self.name, self.version))
         if info[0] is not 0:
             exception = "Error downloading package %s@%s\n" % (self.name, self.version)
             exception += info[1]
             raise ValueError(exception)
         tarball_file = info[1].split('\n')[-1]
+        self.extract_tarball(tarball_file)
+
+    def given_tarball(self):
+        utils.debug(1, "opening %s tarball" % (self.module_name))
+        tarball_file = "../" + self.module_name
```

Comment:
I'm not sure as to what you're trying to accomplish here? module_name is what is passed with create, right?

if I launch `npm2deb create uniq --no-registry` it'd try to unzip `../uniq` Could you clarify?

Reply:
Yes, it does.  One of the earlier (preexisting) steps is to create and cd into a 'module_name' directory: the command is executed from `yourdir/uniq`, but the tarball is in `yourdir/uniq-1.10.tgz`, so we need that `../`.